src/main/java/org/assertj/core/internal/Paths.java

```diff
@@ -377,6 +380,32 @@ public void assertIsNotEmptyFile(AssertionInfo info, Path actual) {
     }
   }
 
+  public void assertHasSameFileSystem(AssertionInfo info, Path actualPath, FileSystem expected) {
+    assertNotNull(info, actualPath);
+    requireNonNull(expected, "The expected file system should not be null");
+
+    FileSystem actual = actualPath.getFileSystem();
+    requireNonNull(actual, "The actual file system should not be null");
```

Comment:
should the assertion succeeds if both actual and expected have null file system?

Reply:
From what I can see, there shouldn't be a case where a Path can have a null file system, since the API contract doesn't specify that this return value can be null ([nio.File](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html#getFileSystem())).

Along with this, most other file operations would throw a NullPointerException by the looks. The only way to achieve this I believe would be to make a custom Path implementation that returned null here, since Paths.get/Path.of binds to the default file system by default.

Reply:
sounds good, let's keep the current behavior